crates/factor-outbound-networking/src/runtime_config.rs

```diff
@@ -0,0 +1,250 @@
+#[cfg(feature = "spin-cli")]
+pub mod spin;
+
+use std::{collections::HashMap, str::FromStr, sync::Arc};
+
+use anyhow::{ensure, Context};
+use rustls::{ClientConfig, RootCertStore};
+use rustls_pki_types::{CertificateDer, PrivateKeyDer};
+
+/// Runtime configuration for outbound networking.
+#[derive(Debug)]
+pub struct RuntimeConfig {
+    // Maps component ID -> HostClientConfigs
+    component_host_client_configs: HashMap<String, HostClientConfigs>,
+    default_client_config: Arc<ClientConfig>,
+}
+
+// Maps host authority -> ClientConfig
+type HostClientConfigs = Arc<HashMap<String, Arc<ClientConfig>>>;
+
+impl RuntimeConfig {
+    pub fn new(tls_configs: impl IntoIterator<Item = TlsConfig>) -> anyhow::Result<Self> {
+        let mut component_host_client_configs = HashMap::<String, HostClientConfigs>::new();
+        for tls_config in tls_configs {
+            ensure!(
+                !tls_config.components.is_empty(),
+                "client TLS 'components' list may not be empty"
+            );
+            ensure!(
+                !tls_config.hosts.is_empty(),
+                "client TLS 'hosts' list may not be empty"
+            );
+            let client_config = Arc::new(tls_config.to_client_config()?);
```

Comment:
The errors coming from `to_client_config` will likely be extremely specific to the `ClientConfig` creation process. At the very least, we should add `.context` call here to let the caller know there was an error when trying to create a tls configuration.